app/src/main/java/org/schabi/newpipe/ReCaptchaActivity.java

```diff
@@ -137,23 +178,54 @@ private void saveCookiesAndFinish() {
     }
 
 
-    private void handleCookies(final String url) {
+    private void handleCookiesFromUrl(final @Nullable String url) {
+        if (MainActivity.DEBUG) {
+            Log.d(TAG, "handleCookiesFromUrl: url=" + (url == null ? "null" : url));
+        }
+
+        if (url == null) {
+            return;
+        }
+
         String cookies = CookieManager.getInstance().getCookie(url);
+        handleCookies(cookies);
+
+        // sometimes cookies are inside the url
+        int abuseStart = url.indexOf("google_abuse=");
+        if (abuseStart != -1) {
+            int abuseEnd = url.indexOf("+path");
+
+            try {
+                String abuseCookie = url.substring(abuseStart + 13, abuseEnd);
+                abuseCookie = URLDecoder.decode(abuseCookie, "UTF-8");
+                handleCookies(abuseCookie);
+            } catch (UnsupportedEncodingException | StringIndexOutOfBoundsException e) {
+                if (MainActivity.DEBUG) {
+                    e.printStackTrace();
+                    Log.d(TAG, "handleCookiesFromUrl: invalid google abuse starting at "
+                            + abuseStart + " and ending at " + abuseEnd + " for url " + url);
```

Comment:
Is that information useful to find further cookies? We might want to show a snackbar to allow reporting the error and more importantly, the url

Reply:
This piece of code has never been useful in my testing, since it just obtains a cookie that would anyway appear as a cookie in one of the subsequent requests. Though to prevent issues under different environment where the webview behaves strangely (who knows?), it doesn't hurt to obtain the cookie twice in a different way and call it a day.

I left a Log there just so that when I send people an apk to debug, I don't have to make a custom one that throws a random Exception with the log message inside (as I did with the other cookie testing apk), but I can just tell them to install a logcat viewer and I have the information needed.

Anyway, if the catch block is ever reached it means the url isn't useful to obtain a cookie, so there is no need to report anything to a normal user imho.

Reply:
However logcat apps need root access since Jelly Bean.

Reply:
Oh, do they? I didn't know ;-)
But what could I implement then, that would log everything and display it to the user after he presses "Ok" on a snackbar? Should I do some kind of bug report, but with logs instead of stack trace?